### PR TITLE
PATCH: 4.0.0-beta-3

### PR DESCRIPTION
**Pull request by jsflax**
Assorted changes:
- Updated incompatible Swift grammars for target `11.0`.
- Added coveralls code coverage support.
- Removed constants in `Info.plist` files. These are now environment variables that must be passed into the `build`/`test` commands.
- Updated Makefile so only the `make` command needs to be ran to prepare the project for development.
- Removed extra dylibs from MongoSwiftMobile and StitchCoreLocalMongoDBService.
- Reorganized scripts.

### CODE REVIEW

File: .evg.yml [hunk 1]
```diff
@@ -152,71 +134,69 @@ tasks:
...
-                ENABLE_BITCODE=NO \
-                IPHONEOS_DEPLOYMENT_TARGET="8.0" \
-                `[[ $no_test != YES ]] && echo "test" || echo ""`
-            }
 
             echo "...testing core"
-            run_xctest StitchCoreSDK StitchCoreSDK-Package
-            run_xctest StitchCoreAdminClient StitchCoreAdminClient-Package
-            run_xctest StitchCoreTestUtils StitchCoreTestUtils-Package
+            ./scripts/run_xctest.sh StitchCoreSDK StitchCoreSDK-Package NO $SIM_UUID
```
Comment (edaniels) [inline .evg.yml]: [opt] given the repetitive nature of this, you could create a bash array containing each module of each section to run the tests on. Would cut down the script size a bit.

File: .evg.yml [hunk 2]
```diff
@@ -152,71 +134,69 @@ tasks:
...
+            ./scripts/run_xctest.sh StitchAWSS3Service StitchAWSS3Service NO $SIM_UUID AWS_ACCESS_KEY_ID=${test_aws_key} AWS_SECRET_ACCESS_KEY=${test_aws_secret}
+            ./scripts/run_xctest.sh StitchAWSSESService StitchAWSSESService NO $SIM_UUID AWS_ACCESS_KEY_ID=${test_aws_key} AWS_SECRET_ACCESS_KEY=${test_aws_secret}
+            ./scripts/run_xctest.sh StitchFCMService StitchFCMService NO $SIM_UUID FCM_SENDER_ID=${test_fcm_sender_id} FCM_API_KEY=${test_fcm_api_key}
+            ./scripts/run_xctest.sh StitchHTTPService StitchHTTPService NO $SIM_UUID
+            ./scripts/run_xctest.sh StitchRemoteMongoDBService StitchRemoteMongoDBService NO $SIM_UUID
+            ./scripts/run_xctest.sh StitchTwilioService StitchTwilioService NO $SIM_UUID TWILIO_SID=${test_twilio_sid} TWILIO_AUTH_TOKEN=${test_twilio_authtoken}
 
             xcrun simctl shutdown $SIM_UUID
             xcrun simctl erase $SIM_UUID
+      - command: shell.exec
```
Comment (edaniels) [inline .evg.yml]: [dope]

File: .gitignore [hunk 3]
```diff
@@ -147,3 +147,6 @@ fastlane/test_output
 
 dist
 vendor
+CoverageData
+localDerivedData
+.vscode
```
Comment (edaniels) [inline .gitignore]: EOL

File: scripts/xccoveralls.py [hunk 4]
```diff
@@ -0,0 +1,305 @@
+import argparse
```
Comment (edaniels) [inline scripts/xccoveralls.py]: fun

File: iOS/Services/StitchAWSSESService/StitchAWSSESServiceTests/AWSSESServiceTestsConstants.h [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#define STRINGIZE(x) #x
```
Comment (edaniels) [inline iOS/Services/StitchAWSSESService/StitchAWSSESServiceTests/AWSSESServiceTestsConstants.h]: this is really cool

File: Core/Services/StitchCoreTwilioService/Makefile [hunk 6]
```diff
@@ -18,3 +18,4 @@ test:
 	xcodebuild test -workspace ../../../Stitch.xcworkspace/ -scheme StitchCoreTwilioService-Package -configuration Debug -destination "platform=iOS Simulator,name=iPhone 7,OS=11.2"
 project:
 	swift package --build-path ../../../.build generate-xcodeproj --xcconfig-overrides StitchCoreTwilioService.xcconfig
+prepare: git resolve project
```
Comment (edaniels) [inline Core/Services/StitchCoreTwilioService/Makefile]: EOL

File: Core/Services/StitchCoreLocalMongoDBService/Makefile [hunk 7]
```diff
@@ -18,3 +18,4 @@ test:
 	xcodebuild test -workspace ../../../Stitch.xcworkspace/ -scheme StitchCoreLocalMongoDBService-Package -configuration Debug -destination "platform=iOS Simulator,name=iPhone 7,OS=11.2"
 project:
 	swift package --build-path ../../../.build generate-xcodeproj --xcconfig-overrides StitchCoreLocalMongoDBService.xcconfig
+prepare: git resolve project
```
Comment (adamchel) [inline Core/Services/StitchCoreLocalMongoDBService/Makefile]: [nit] newline

**Pull request merged**